pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -209,11 +209,17 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 	}
 
 	before := pr.Status.GetCondition(duckv1alpha1.ConditionSucceeded)
-	after := resources.GetPipelineConditionStatus(pr.Name, state, c.Logger)
+	after := resources.GetPipelineConditionStatus(pr.Name, pipelineState, c.Logger)
 	pr.Status.SetCondition(after)
 
 	reconciler.EmitEvent(c.Recorder, before, after, pr)
 
+	for _, pipelineItem := range pipelineState {
+		if pipelineItem.TaskRun != nil {
+			pr.Status.TaskRuns[pipelineItem.TaskRun.Name] = pipelineItem.TaskRun.Status
+		}
+	}
```

Comment:
can we add some unit test coverage for this? it'll mean breaking it out into a function(s)

Reply:
its covered in the pipelinerun_test which is checking the status of the PipelineRun, but i'll look into adding more coverage